### ISSUE TITLE
Upstream constructor changes from fork to project

### ISSUE THREAD

**Issue by goanpeca** [napari/packaging#15]
## 🧰 Task

For the constructor bundle work @jaimergp used a fork and implemented a bunch if goodies.

The plan is to upstream this from the work https://github.com/jaimergp/constructor/tree/menuinst+branding

**Comment by potating-potato** [comment]
@jaimergp curious about the next steps on this issue, are these blocking our migration or?

**Comment by jaimergp** [comment]
Not blocking. Most of the constructor-only stuff landed on `main`, but the `menuinst` derived features are still pending.

**Comment by larsoner** [comment]
@jaimergp don't forget to check off https://github.com/conda/constructor/pull/475 :)

**Comment by jaimergp** [comment]
Adding some task items here for my own sanity as I update I sync our fork to latest upstream versions to reduce the technical debt. We should have 90% parity or so!

- [ ] Cut a new `menuinst` alpha release from https://github.com/conda/menuinst/pull/91
- [ ] Cut a new `constructor` build from https://github.com/conda/constructor/pull/474
- [ ] Cut a new `conda` build from https://github.com/conda/conda/pull/11882
- [ ] Build a new `conda-standalone` using the three PRs mentioned above
- [ ] Upload builds to a new label `bundle-tools-2` to not disrupt established workflows
- [ ] Update workflows to use the new label
- [ ] Test everything works and if #38 is fixed

**Comment by jaimergp** [comment]
The above is not done yet, but we are going to update our installers to use the last features in `menuinst`.  This is the build order:

- [X] Create new channel label: `napari/label/bundle_tools_3`
- [ ] Release new menuinst build 
- [ ] Update from `main` and release new conda build
- [ ] Update from `main` and release new constructor build
- [ ] Release new conda-standalone build
- [ ] Upgrade to `bundle_tools_3`
  - [ ] Build new installers
  - [ ] Test new installers

**Comment by jaimergp** [comment]
A last update on this:

* We released menuinst v2 in November I think? We have released two patch versions since then. We have 2.0.2 right now!
* conda 23.11.0 (released in December) ships compatibility for menuinst v2
* constructor 3.6.0 shipped compatibility with menuinst v2 last week
* conda-standalone 23.11.0 with, you guessed it, menuinst v2 compatibility, has just been merged! We only need a [core review to package the manual osx-arm64 build](https://github.com/conda-forge/conda-standalone-feedstock/pull/69).
* https://github.com/napari/packaging/pull/103 drops `napari/label/bundle_tools_*` and uses the conda-canary nightlies. I just need to adjust it so it uses conda-forge straight once all the releases land.

So, all in all, it sounds like we are basically done! This won't reach our installers till napari v0.5 is released, though. The 0.4.19 pipeline will still use the "old yet trusted" forked version in `main`. Once that lands, we can merge all the open PRs and upgrade to the upstream stack! We'll need some QA at some point to ensure everything is ✨.

In the meantime, I'll send a couple PRs to update our packaging docs. An by update I mean "dramatically simplify" because now we can just omit all the hacky details of our fork.

It only took use 2.5 years 🚀
